numpy/core/src/multiarray/array_method.c

```diff
@@ -334,28 +334,29 @@ fill_arraymethod_from_slots(
     }
 
     /* Check whether the provided loops make sense. */
-    if (meth->strided_loop == NULL) {
+    if ((meth->unaligned_strided_loop == NULL) !=
+            !(meth->flags & NPY_METH_SUPPORTS_UNALIGNED)) {
         PyErr_Format(PyExc_TypeError,
-                "Must provide a strided inner loop function. (method: %s)",
+                "Must provide unaligned strided inner loop for method to "
+                "indicate unaligned support (method: %s)",
```

Comment:
It might be a bit clearer to dtype authors if you had a different error message for the case when someone defines an `unaligned_strided_loop` but doesn't set the flag correctly.